packages/run-protocol/src/metrics.js

```diff
@@ -0,0 +1,40 @@
+// @ts-check
+import { makeStoredSubscription, makeSubscriptionKit } from '@agoric/notifier';
+import { E } from '@endo/eventual-send';
+
+const PATH = 'metrics';
+
+/**
+ * @template T
+ * @param {ERef<StorageNode>} [storageNode]
+ * @param {ERef<Marshaller>} [marshaller]
+ * @param {T[]} optionalInitialState
```

Comment:
Not specifying an initial state is a behaviour @dtribble is actively trying to stamp out.  The new `makePublisherKit` will not support it.

So, would you please change this not to be optional, and have the call sites always supply it?
```suggestion
 * @param {T} initialState
```

Reply:
This constraint greatly complicates virtual object initial state because the publisher is part of that state, but would not also require that state defined (minus the publisher) in order to create the publisher.

Why is @dtribble trying to stamp it out? Is there any other reason that `makePublisherKit` will not support it?

Reply:
@erights argued against requiring an initial state. I have arguments for it. I don't recall where we ended up. Much of that is in an IBIS.

Reply:
I think that if initialState is optional, then we should not support it at all. It's better to just explicitly publish if you want

Reply:
https://github.com/Agoric/agoric-sdk/pull/5621